lib/modules/ai_bot/commands/setting_context_command.rb

```diff
@@ -0,0 +1,148 @@
+#frozen_string_literal: true
+
+module DiscourseAi::AiBot::Commands
+  MAX_CONTEXT_TOKENS = 2000
+
+  class SettingContextCommand < Command
+    def self.rg_installed?
+      @rg_installed ||=
+        begin
+          Discourse::Utils.execute_command("which", "rg")
+          true
+        rescue Discourse::Utils::CommandError
+          false
+        end
+    end
+
+    class << self
+      def name
+        "setting_context"
+      end
+
+      def desc
+        "Will provide you with full context regarding a particular site setting in Discourse"
+      end
+
+      def parameters
+        [
+          Parameter.new(
+            name: "setting_name",
+            description: "The name of the site setting we need context for",
+            type: "string",
+            required: true,
+          ),
+        ]
+      end
+    end
+
+    def result_name
+      "context"
+    end
+
+    def description_args
+      { setting_name: @setting_name }
+    end
+
+    def process(setting_name:)
+      if !self.class.rg_installed?
+        return(
+          {
+            setting_name: setting_name,
+            context: "This command requires the rg command line tool to be installed on the server",
+          }
+        )
+      end
+
+      @setting_name = setting_name
+      if !SiteSetting.has_setting?(setting_name)
+        { setting_name: setting_name, context: "This setting does not exist" }
+      else
+        description = SiteSetting.description(setting_name)
+        result = +"# #{setting_name}\n#{description}\n\n"
+
+        setting_info =
+          find_setting_info(setting_name, [Rails.root.join("config", "site_settings.yml").to_s])
+        if !setting_info
+          setting_info =
+            find_setting_info(setting_name, Dir[Rails.root.join("plugins/**/settings.yml")])
+        end
+
+        result << setting_info
+        result << "\n\n"
+
+        %w[lib app plugins].each do |dir|
+          path = Rails.root.join(dir).to_s
+          result << Discourse::Utils.execute_command(
+            "rg",
+            setting_name,
+            path,
+            "-g",
+            "!**/spec/**",
+            "-g",
+            "!**/dist/**",
+            "-g",
+            "*.{rb,js,gjs}",
```

Comment:
Should probably include `.hbs` files here too?

Reply:
can we have site settings in hbs files?

Reply:
Yes we definitely can, client side ones anyway. Eg you can do this.siteSettings.blah in a component